src/bindings/wasm/tvgWasmLottieAnimation.cpp

```diff
@@ -35,13 +38,58 @@ class __attribute__((visibility("default"))) TvgLottieAnimation
 public:
     ~TvgLottieAnimation()
     {
-        free(buffer);
+        if (engine == tvg::CanvasEngine::Sw) {
+        #ifdef THORVG_SW_RASTER_SUPPORT
+            free(buffer);
+        #endif
+        } else if (engine == tvg::CanvasEngine::Wg) {
+        #ifdef THORVG_WG_RASTER_SUPPORT
+            wgpuSurfaceRelease(surface);
+            wgpuInstanceRelease(instance);
+        #endif
+        }
+
         Initializer::term();
     }
 
-    static TvgLottieAnimation* create()
+    explicit TvgLottieAnimation(string engine = "sw", string selector = "")
     {
-        return new TvgLottieAnimation;
+        errorMsg = NoError;
+
+        if (engine == "sw") {
```

Comment:
59 ~ 71 looks unnecessary.

Reply:
@hermet Can we generate canvas before `init()`?

I was thiking `Initializer::init(0, this->engine)` can't be called if we move the line into canvas creation part.

Reply:
Yes, it should be. You can move it. I believe even it's compact and simpler code.